### PR TITLE
[BugFix] Fix small file related security issues

### PR DESCRIPTION
**Pull request by nshangyiming**
Why I'm doing:
We have found some potential security issues when reviewing the http api.
Severity: Low.

What I'm doing:
- Santize some parameters of http api to avoid leaking credentials to fe.log.
- Add authorization to get_log_file api.
- Make the error message of `/bootstrap` api generic to avoid token sniffing.
- Clean some code warnings.

## What type of PR is this:

- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.2
  - [x] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

**Review by HangyuanLiu — DISMISSED** [review]

File: fe/fe-core/src/main/java/com/starrocks/http/HttpServerHandler.java [hunk 1]
```diff
@@ -150,7 +151,7 @@ private void writeResponse(ChannelHandlerContext context, HttpResponseStatus sta
     }
 
     @Override
-    public void exceptionCaught(ChannelHandlerContext ctx, Throwable cause) throws Exception {
+    public void exceptionCaught(ChannelHandlerContext ctx, Throwable cause) {
```
Comment (kevincai) [inline fe/fe-core/src/main/java/com/starrocks/http/HttpServerHandler.java:154]: A tiny improvement here, no need to be part of this PR

add debug log to print the exception stack. Several times we saw http server excepted, but didn't know where is the exception point, it is nice to be able to turn on debug log here to log the stack trace for easier debug and fix.

File: fe/fe-core/src/main/java/com/starrocks/http/rest/BootstrapFinishAction.java [hunk 2]
```diff
@@ -87,20 +91,24 @@ public void execute(BaseRequest request, BaseResponse response) throws DdlExcept
...
-                    result.msg = "invalid cluster id format: " + clusterIdStr;
+                    LOG.info("invalid cluster id format: {}", clusterIdStr);
+                    result.msg = "invalid parameter";
                 }
 
                 if (result.status == ActionStatus.OK) {
                     if (clusterId != GlobalStateMgr.getCurrentState().getNodeMgr().getClusterId()) {
                         result.status = ActionStatus.FAILED;
-                        result.msg = "invalid cluster id: " + GlobalStateMgr.getCurrentState().getNodeMgr().getClusterId();
+                        LOG.info("invalid cluster id: {}",
```
Comment (kevincai) [inline fe/fe-core/src/main/java/com/starrocks/http/rest/BootstrapFinishAction.java:101]: should log the invalid cluster id, not the correct cluster id?

File: fe/fe-core/src/main/java/com/starrocks/http/rest/BootstrapFinishAction.java [hunk 3]
```diff
@@ -87,20 +91,24 @@ public void execute(BaseRequest request, BaseResponse response) throws DdlExcept
...
+                                GlobalStateMgr.getCurrentState().getNodeMgr().getClusterId());
+                        result.msg = "invalid parameter";
                     }
                 }
 
                 if (result.status == ActionStatus.OK) {
                     if (!token.equals(GlobalStateMgr.getCurrentState().getNodeMgr().getToken())) {
                         result.status = ActionStatus.FAILED;
-                        result.msg = "invalid token: " + GlobalStateMgr.getCurrentState().getNodeMgr().getToken();
+                        LOG.info("invalid token: {}", GlobalStateMgr.getCurrentState().getNodeMgr().getToken());
```
Comment (kevincai) [inline fe/fe-core/src/main/java/com/starrocks/http/rest/BootstrapFinishAction.java:110]: similiar, it is the `token` invalid, not the `GlobalStateMgr....getToken()` invalid.

File: fe/fe-core/src/main/java/com/starrocks/http/WebUtils.java [hunk 4]
```diff
@@ -161,4 +171,48 @@ private static char decodeHexNibble(final char c) {
...
+    /**
+     * This method sanitizes the input URI by replacing the values of certain query parameters with "*".
+     * The parameters to be sanitized are listed in `SANITIZED_HTTP_PARAMS`.
+     * If these parameters are not present in the URI, the method returns the original URI.
+     *
+     * @param uri The input URI to be sanitized.
+     * @return The sanitized URI with the values of certain query parameters replaced with "*".
+     * @throws URISyntaxException If the input string can't be parsed as a URI reference.
+     */
+    public static String sanitizeHttpReqUri(String uri) throws URISyntaxException {
```
Comment (kevincai) [inline fe/fe-core/src/main/java/com/starrocks/http/WebUtils.java:184]: wondering if can leverage QueryStringDecoder to simplify some code.

Comment (nshangyiming) [reply fe/fe-core/src/main/java/com/starrocks/http/WebUtils.java:184]: not sure, chatgpt told me to do it this way...

File: fe/fe-core/src/main/java/com/starrocks/http/HttpServerHandler.java (hunk 1 above)
Comment (kevincai) [reply fe/fe-core/src/main/java/com/starrocks/http/HttpServerHandler.java:154]: https://github.com/StarRocks/starrocks/issues/40316 @nshangyiming

Comment (nshangyiming) [reply fe/fe-core/src/main/java/com/starrocks/http/HttpServerHandler.java:154]: copy that

File: fe/fe-core/src/main/java/com/starrocks/http/rest/BootstrapFinishAction.java (hunk 2 above)
Comment (nshangyiming) [reply fe/fe-core/src/main/java/com/starrocks/http/rest/BootstrapFinishAction.java]: fixed

File: fe/fe-core/src/main/java/com/starrocks/http/rest/BootstrapFinishAction.java (hunk 3 above)
Comment (nshangyiming) [reply fe/fe-core/src/main/java/com/starrocks/http/rest/BootstrapFinishAction.java]: fixed

**Review by kevincai — APPROVED** [review]

**Review by HangyuanLiu — APPROVED** [review]

**Pull request merged**